product_pricelist_supplierinfo/models/product_supplierinfo.py

```diff
@@ -11,15 +10,11 @@ class ProductSupplierinfo(models.Model):
     sale_margin = fields.Float(
         'Sale Margin', default=0, digits=(16, 2),
         help="Margin to apply on price to obtain sale price")
-    sale_price = fields.Float(
-        compute='_compute_product_price',
-        store=True,
-        digits=dp.get_precision('Product Price'),
-        help="Sale price used in sale pricelist computed from supplierinfo")
 
-    @api.depends('price', 'sale_margin')
-    def _compute_product_price(self):
-        for product in self:
-            product.sale_price = (
-                (product.price + (product.price * (product.sale_margin / 100)))
-                or 0.0)
+    @api.multi
+    def _get_supplierinfo_pricelist_price(self):
+        self.ensure_one()
+        sale_price = (
```

Comment:
Please don't do the math if sale_margin == 100

Reply:
...or 0

Reply:
Ok